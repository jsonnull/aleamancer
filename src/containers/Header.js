// @flow
import React from 'react'
import { connect } from 'react-redux'
import { push } from 'react-router-redux'
import Header from 'components/Header'
import { SHOW_SETTINGS } from 'actions/types'
import type { State } from 'store'

type StateProps = {
  username: string
}
const mapStateToProps = (state: State): StateProps => ({
  username: state.user.profile.displayName
})

type DispatchProps = {
  goHome: Function,
  showSettings: Function
}
const mapDispatchToProps = (dispatch: Function): DispatchProps => ({
  goHome: () => dispatch(push('/')),
  showSettings: () => dispatch({ type: SHOW_SETTINGS })
})

export default connect(mapStateToProps, mapDispatchToProps)(Header)
