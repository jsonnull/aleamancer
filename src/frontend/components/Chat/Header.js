// @flow
import React from 'react'
import styled from 'styled-components'
import Tooltip from '../Tooltip'
import { fontSize } from 'frontend/styles/common'

const Container = styled.div`
  background-color: ${props =>
    props.isPinned ? props.theme.background : props.theme.backgroundInverted};
  display: flex;
  flex-direction: row;
`

const Toggle = styled.div`
  display: block;
  font-size: ${fontSize.small};
  color: ${props =>
    props.isPinned ? props.theme.color : props.theme.colorInverted};
  margin-right: 1rem;
  padding-left: 1rem;
  cursor: pointer;

  i {
    transform: ${props => (props.isPinned ? 'none' : 'rotate(45deg)')};
  }
`

const Outer = styled.div`
  margin-left: auto;
`

type Props = {
  isPinned: boolean,
  setChatPinned: Function
}

const Header = (props: Props) => {
  const { isPinned, setChatPinned } = props

  let toggleChat = isPinned ? (
    <i className="fa fa-chevron-left" />
  ) : (
    <i className="fa fa-thumb-tack" />
  )

  const tooltip = isPinned ? 'Unpin Chat' : 'Pin Chat'

  return (
    <Container isPinned={isPinned}>
      <Outer>
        <Tooltip placement="left" content={tooltip}>
          <Toggle onClick={() => setChatPinned(!isPinned)} isPinned={isPinned}>
            {toggleChat}
          </Toggle>
        </Tooltip>
      </Outer>
    </Container>
  )
}

export default Header
