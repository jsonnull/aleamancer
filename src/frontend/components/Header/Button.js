// @flow
import styled from 'styled-components'
import { fontSize } from 'frontend/styles/common'

const Button = styled.div`
  height: 3.25rem;
  width: ${props => (props.square ? '3.25rem' : 'auto')};
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: ${fontSize.normal};
  background: ${props => props.theme.backgroundInverted};
  cursor: pointer;
  border-radius: 5px;
  color: ${props => props.theme.colorInverted};

  &:hover {
    background: ${props => props.theme.backgroundInvertedSecondary};
  }
`

export default Button
