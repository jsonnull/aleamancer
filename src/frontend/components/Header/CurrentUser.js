// @flow
import React from 'react'
import { fontSize } from 'frontend/styles/common'
import Button from './Button'

const SmallButton = Button.extend`
  font-size: ${fontSize.small};
`

type Props = {
  username: string
}

const CurrentUser = (props: Props) => {
  return <SmallButton>{props.username}</SmallButton>
}

export default CurrentUser
