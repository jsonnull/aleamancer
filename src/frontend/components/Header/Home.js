// @flow
import React from 'react'
import { Link } from 'react-router-dom'
import Tooltip from '../Tooltip'
import Button from './Button'

const Home = () => (
  <Tooltip placement="right" content="Home">
    <Link to="/sessions">
      <Button square>
        <i className="fa fa-home" />
      </Button>
    </Link>
  </Tooltip>
)

export default Home
