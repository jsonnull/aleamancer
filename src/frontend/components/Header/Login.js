// @flow
import React from 'react'
import { Link } from 'react-router-dom'
import Button from './Button'

const Login = () => (
  <Link to="/login" style={{ textDecoration: 'none' }}>
    <Button>Login</Button>
  </Link>
)

export default Login
