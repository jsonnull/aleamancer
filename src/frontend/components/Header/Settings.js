// @flow
import React from 'react'
import Button from './Button'

type Props = {
  showSettings: Function
}

const Settings = (props: Props) => {
  const { showSettings } = props

  return (
    <Button square onClick={showSettings}>
      <i className="fa fa-cog" />
    </Button>
  )
}

export default Settings
