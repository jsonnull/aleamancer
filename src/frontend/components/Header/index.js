// @flow
import * as React from 'react'
import styled from 'styled-components'
import { CONSTS } from 'frontend/styles/common'
import Home from './Home'
import Logo from './Logo'
import CurrentUser from './CurrentUser'
import Settings from './Settings'
import Login from './Login'

const Container = styled.div`
  display: flex;
  flex-direction: row;
  height: ${CONSTS.headerHeight};
  line-height: ${CONSTS.headerHeight};
  color: white;
  background-color: ${props => props.theme.backgroundInverted};
`

const Column = styled.div`
  display: flex;
  flex-direction: row;
  flex: 1;
  padding: 0.5rem;
`

const Right = Column.extend`
  justify-content: flex-end;
`

type Props = {
  username: string | null,
  showSettings: Function
}

const Header = (props: Props) => {
  const { username, showSettings } = props

  return (
    <Container>
      <Column>
        <Home />
      </Column>
      <Column>
        <Logo />
      </Column>
      <Right>
        {username === null ? (
          <Login />
        ) : (
          <React.Fragment>
            <CurrentUser username={username} />
            <Settings showSettings={showSettings} />
          </React.Fragment>
        )}
      </Right>
    </Container>
  )
}

export default Header
