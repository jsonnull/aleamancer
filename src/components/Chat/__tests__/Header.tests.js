// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import Header from '../Header.js'

describe('Chat header component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Header isPinned={false} toggleChatPin={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
