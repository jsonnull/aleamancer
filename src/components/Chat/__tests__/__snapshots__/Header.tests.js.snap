// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat header component renders correctly 1`] = `
<div
  className="sc-bdVaJa dMsinD"
>
  <div
    className="sc-bwzfXH ellkeL"
    onClick={[Function]}
  >
    <i
      className="fa fa-thumb-tack"
    />
  </div>
</div>
`;
